File: java/json-transform/src/main/java/co/nlighten/jsontransform/template/TextTemplateJsonTransformer.java
package co.nlighten.jsontransform.template;

import co.nlighten.jsontransform.JsonTransformer;
import co.nlighten.jsontransform.ParameterResolver;
import co.nlighten.jsontransform.adapters.JsonAdapter;

public class TextTemplateJsonTransformer extends JsonTransformer {
    public TextTemplateJsonTransformer(JsonAdapter<?, ?, ?> adapter) {
        super(null, adapter);
    }

    public Object transformString(Object definition, ParameterResolver resolver) {
        return fromJsonPrimitive("$", definition, resolver, false);
    }
}

File: javascript/json-transform/src/template/TextTemplateJsonTransformer.ts
import { JsonTransformer } from "../JsonTransformer";
import { ParameterResolver } from "../ParameterResolver";
import { TextTemplateJsonTransformFunction } from "./TextTemplateJsonTransformFunction";

class TextTemplateJsonTransformer extends JsonTransformer {
  constructor() {
    super(null);
  }
  public transformString(definition: any, resolver: ParameterResolver) {
    return super.fromJsonPrimitive("$", definition, resolver, false);
  }
}

const ttjt = new TextTemplateJsonTransformer();
export const TextTemplateJsonTransformFunctionInstance: TextTemplateJsonTransformFunction =
  ttjt.transformString.bind(ttjt);
